src/Catalyst.Core.Modules.Ledger/Web3EthApi.cs

```diff
@@ -22,25 +22,36 @@
 #endregion
 
 using System;
+using Autofac.Features.AttributeFilters;
+using Catalyst.Abstractions.Consensus.Deltas;
 using Catalyst.Abstractions.Kvm;
 using Catalyst.Abstractions.Ledger;
+using Nethermind.Evm;
 using Nethermind.Store;
 
 namespace Catalyst.Core.Modules.Ledger 
 {
     public class Web3EthApi : IWeb3EthApi
-    {
-        public Web3EthApi(IDeltaExecutor deltaExecutor, IStateReader stateReader, IStateRootResolver stateRootResolver, IDeltaResolver deltaResolver)
+    { 
+        public const string ComponentName = nameof(Web3EthApi);
+            
+        public Web3EthApi(IStateReader stateReader, IDeltaResolver deltaResolver, IDeltaCache deltaCache, [KeyFilter(ComponentName)] ITransactionProcessor processor, [KeyFilter(ComponentName)] IStorageProvider storageProvider, [KeyFilter(ComponentName)] IStateProvider stateProvider)
         {
-            DeltaExecutor = deltaExecutor ?? throw new ArgumentNullException(nameof(deltaExecutor));
             StateReader = stateReader ?? throw new ArgumentNullException(nameof(stateReader));
-            StateRootResolver = stateRootResolver ?? throw new ArgumentNullException(nameof(stateRootResolver));
             DeltaResolver = deltaResolver ?? throw new ArgumentNullException(nameof(deltaResolver));
+            DeltaCache = deltaCache ?? throw new ArgumentNullException(nameof(deltaCache));
+            Processor = processor ?? throw new ArgumentNullException(nameof(processor));
+            StorageProvider = storageProvider ?? throw new ArgumentNullException(nameof(storageProvider));
+            StateProvider = stateProvider ?? throw new ArgumentNullException(nameof(stateProvider));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The parameter name '' is not declared in the argument list.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4585091)